### PR TITLE
Require package readmes for shipping packages

### PR DESCRIPTION
**Pull request by ViktorHofer**
Contributes to https://github.com/dotnet/runtime/issues/99358

1. Commit: Require package readmes for shipping packages
Add a target that validates that a package readme
exists for a shipping package.
Set EnableDefaultPackageReadmeFile=false in projects
that don't yet have a package readme.
2. Commit: Delete empty package readme files that were accidentally committed

I will send out a mail to the runtimerepo DL after this change is merged to explain that package readmes are now required and 
to create energy on adding the remaining 30 package READMEs, ideally ready for .NET 9.

### CODE REVIEW

**Comment by am11** [comment]
I noticed that the `packs` subset also reports some missing readmes:

```
2024-07-15T09:43:53.9976756Z   The package Microsoft.NETCore.App.Ref.9.0.0-dev is missing a readme. Go to https://aka.ms/nuget/authoring-best-practices/readme to learn why package readmes are important.
2024-07-15T09:43:53.9976756Z   The package Microsoft.NETCore.App.Ref.9.0.0-dev is missing a readme. Go to https://aka.ms/nuget/authoring-best-practices/readme to learn why package readmes are important.
2024-07-15T09:44:32.1884036Z   The package Microsoft.NETCore.App.Crossgen2.freebsd-x64.9.0.0-dev is missing a readme. Go to https://aka.ms/nuget/authoring-best-practices/readme to learn why package readmes are important.
2024-07-15T09:44:55.9404147Z   The package Microsoft.NETCore.App.Runtime.freebsd-x64.9.0.0-dev is missing a readme. Go to https://aka.ms/nuget/authoring-best-practices/readme to learn why package readmes are important.
2024-07-15T09:45:17.6289662Z   The package Microsoft.NET.HostModel.9.0.0-dev is missing a readme. Go to https://aka.ms/nuget/authoring-best-practices/readme to learn why package readmes are important.
```

**Comment by ViktorHofer** [comment]
@am11 I changed a property so that this validation is only enabled for src/libraries packages (for now).

**Comment by am11** [comment]
Sounds good. I think the validation is probably hitting false-positive, because those are internal packages and nuget.org shows this line: https://github.com/dotnet/runtime/blob/66ae90f3b7ec4f13fffcd71913eca0b45777e58c/Directory.Build.targets#L77 e.g. https://www.nuget.org/packages/Microsoft.NETCore.App.Crossgen2.linux-x64. In these cases, we probably don't need a physical readme file.

**Comment by ViktorHofer** [comment]
> Sounds good. I think the validation is probably hitting false-positive, because those are internal packages and nuget.org shows this line:

Yes we can revisit this if we want to ever expand the validation to other packages produced in runtime, outside of src/libraries.

**Comment by ViktorHofer** [comment]
@dotnet/runtime-infrastructure, @ericstj  would you mind reviewing?

File: eng/packaging.targets
Comment (carlossanlop) [inline eng/packaging.targets:24]: I felt like this was redundant.

```suggestion
    <!-- PackageReadmeFile specifies the package readme file name in the package. PackageReadmeFilePath points to the package readme file on disk. -->
```

File: eng/packaging.targets
Comment (carlossanlop) [inline eng/packaging.targets:26]: Why doesn't the value contain a full path or a relative path? It's just the filename, but the property name indicates otherwise.

File: eng/packaging.targets
Comment (carlossanlop) [inline eng/packaging.targets:91]: This comment is strange. From where?

File: eng/packaging.targets
Comment (carlossanlop) [inline eng/packaging.targets:93]: Is this where the path is prefixed? Maybe PackageReadmeFilePath should be named PackageReadmeFileName?

File: eng/packaging.targets
Comment (carlossanlop) [inline eng/packaging.targets:320]: I suppose this is all relative to the project that is invoking the target, hence why README.md does not really need a prefix path?

File: src/libraries/Directory.Build.targets
Comment (carlossanlop) [inline src/libraries/Directory.Build.targets:131]: In which CI leg do we enable IsShipping?

File: src/libraries/System.IO.Ports/pkg/runtime.native.System.IO.Ports.props
Comment (carlossanlop) [inline src/libraries/System.IO.Ports/pkg/runtime.native.System.IO.Ports.props:17]: Why do we have a secondary definition of this property in a props file, besides the one in the csproj?

File: src/libraries/System.IO.Ports/pkg/runtime.native.System.IO.Ports.proj
Comment (carlossanlop) [inline src/libraries/System.IO.Ports/pkg/runtime.native.System.IO.Ports.proj:10]: Do you mind explaining why this is different? I see it's from a pkg folder, which is rare for a library to have.

**Review by carlossanlop — COMMENTED** [review]
Looks good. I have questions.

File: eng/packaging.targets
Comment (ViktorHofer) [reply eng/packaging.targets:26]: The path is relative to the project file which is the default when including stuff in msbuild.

File: src/libraries/Directory.Build.targets
Comment (ViktorHofer) [reply src/libraries/Directory.Build.targets:131]: IsShipping defaults to true and some projects (ie transport ones) set it to false.

File: src/libraries/System.IO.Ports/pkg/runtime.native.System.IO.Ports.props
Comment (ViktorHofer) [reply src/libraries/System.IO.Ports/pkg/runtime.native.System.IO.Ports.props:17]: The .props file is for all the runtime.RID packages (i.e. https://www.nuget.org/packages/runtime.osx-x64.runtime.native.System.IO.Ports/) and the proj one is the runtime.native project: https://www.nuget.org/packages/runtime.native.System.IO.Ports/9.0.0-preview.6.24327.7

File: src/libraries/System.IO.Ports/pkg/runtime.native.System.IO.Ports.proj
Comment (ViktorHofer) [reply src/libraries/System.IO.Ports/pkg/runtime.native.System.IO.Ports.proj:10]: System.IO.Ports consists of:
1. A NuGet package for the lib -> https://www.nuget.org/packages/System.IO.Ports/9.0.0-preview.6.24327.7
2. A NuGet package for the native meta package -> https://www.nuget.org/packages/runtime.native.System.IO.Ports/
3. A NuGet package for the native RID specific package -> https://www.nuget.org/packages/runtime.linux-arm.runtime.native.System.IO.Ports/

The native packages should just re-use the package README from the lib package.

File: eng/packaging.targets
Comment (ViktorHofer) [reply eng/packaging.targets:320]: Yes. That path in MSBuild is relative to the project file.

File: eng/packaging.targets
Comment (ViktorHofer) [reply eng/packaging.targets:93]: Nope. This is just where the package readme file gets added as a package file so that it gets copied. The `PackageReadmeFile` property determines the path of the file in the package. This is a NuGet extension point, which is weird. This could have been implemented in a better way.

File: src/libraries/System.IO.Ports/pkg/runtime.native.System.IO.Ports.props
Comment (carlossanlop) [reply src/libraries/System.IO.Ports/pkg/runtime.native.System.IO.Ports.props:17]: Okay good, so this ensures we have one readme to rule them all.

**Review by carlossanlop — APPROVED** [review]
LGTM. Consider my comment suggestions optional. Feel free to wait for a review from Eric too.

**Pull request merged**
